src/frontend/js/pages/TeacherTrainingDashboardLoader/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { FormattedMessage, defineMessages, useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+
+import { capitalize } from 'lodash-es';
+import CourseRunList from 'pages/TeacherCourseDashboardLoader/CourseRunList';
+import { DashboardLayout } from 'widgets/Dashboard/components/DashboardLayout';
+import { TeacherCourseDashboardSidebar } from 'widgets/Dashboard/components/TeacherCourseDashboardSidebar';
+import { Spinner } from 'components/Spinner';
+import { DashboardCard } from 'widgets/Dashboard/components/DashboardCard';
+import { Icon, IconTypeEnum } from 'components/Icon';
+import Banner, { BannerType } from 'components/Banner';
+import { useCourseProductRelation } from 'hooks/useCourseProductRelation';
+
+const messages = defineMessages({
+  pageTitle: {
+    defaultMessage: 'Training area',
+    description: 'Use for the page title of the training area',
+    id: 'components.TeacherTrainingDashboardLoader.pageTitle',
+  },
+  loading: {
+    defaultMessage: 'Loading training...',
+    description: 'Message displayed while loading a course',
+    id: 'components.TeacherTrainingDashboardLoader.loading',
+  },
+  errorNoProduct: {
+    defaultMessage: "This product doesn't exist",
+    description: 'Message displayed when requested product is not found',
+    id: 'components.TeacherTrainingDashboardLoader.errorNoProduct',
+  },
+});
+
+export const TeacherTrainingDashboardLoader = () => {
```

Comment:
IMO This page is more than a simple loader, it also displays the page itself, you could create a new separate component to split responsabilities ( like in `DashboardOrderLoader` )